.github/workflows/build.yml

```diff
@@ -0,0 +1,46 @@
+name: Build Sherlock Platform
+
+on:
+  # Triggers the workflow on push or pull request events but only for the idea/242.21829.142 branch
+  push:
+    branches: [ idea/242.21829.142 ]
+  pull_request:
+    branches: [ idea/242.21829.142 ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - name: Set JAVA_HOME
+        run: echo "JAVA_HOME=$JAVA_HOME" >> $GITHUB_ENV
+
+      - name: Grant execute permission to build script
+        run: chmod +x ./build_sherlock_platform.sh
```

Comment:
Do we absolutely need this? AFAIU, the file already has `x` permission. (e.g., did you hit an error without it?)

Reply:
We don't have `+x` permission, but just a shebang. I think it's necessary to run the script on the runner.

Reply:
I tried: `git update-index --chmod=+x build_sherlock_platform.sh` and then `git status` says nothing to commit.

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/adding-scripts-to-your-workflow

Reply:
I'm OK leaving it in for now. (It just seemed redundant.)